src/scripts/circleci-job-canceller/long_job_canceller.py

```diff
@@ -60,7 +79,23 @@ def find_old_workflow_ids(repo_slug, window_start, window_end, headers):
                     yield {"job_status": "too_old", "name": current_workflow['name'], "id": current_workflow['id'], "username": username}
 
 
-def main(circleapitoken, orgreposlug, output_file, commit):
+def do_we_care_about_this_pipeline(current_info, ignore, headers):
+    if ignore == "":
+        return True
+
+    pending_approvals = get_workflow_pending_approval_jobs(current_info['id'], headers)
+
+    # we want to know if all of the names of the approval jobs waiting contain the ignore keyword
+    # if all of them do, then we do NOT care about this pipeline
+    not_ignored_jobs = itertools.dropwhile( lambda x: x['name'].find(ignore) > -1 , pending_approvals )
```

Comment:
can flip this to say whether _any_ job exists which _doesn't_ match the `ignore`